Library/ViewModels/AddNewCardViewModelTests.swift

```diff
@@ -177,64 +180,52 @@ internal final class AddNewCardViewModelTests: TestCase {
 
     self.cardholderNameBecomeFirstResponder
       .assertValueCount(1, "Cardholder name field is first responder when view loads.")
     self.paymentDetailsBecomeFirstResponder.assertDidNotEmitValue("Not first responder when view loads")
     self.zipcodeTextFieldBecomeFirstResponder.assertDidNotEmitValue()
     self.vm.inputs.cardholderNameChanged("")
     self.vm.inputs.paymentInfo(isValid: false)
-    self.vm.inputs.cardBrand(isValid: false)
     self.vm.inputs.zipcodeChanged(zipcode: "")
-    self.saveButtonIsEnabled.assertValues([false], "Disabled form is incomplete")
 
     self.vm.inputs.cardholderNameChanged("Native Squad")
     self.vm.inputs.cardholderNameTextFieldReturn()
     self.cardholderNameBecomeFirstResponder
       .assertValueCount(1, "Does not emit again.")
     self.paymentDetailsBecomeFirstResponder
       .assertValueCount(1, "First responder after editing cardholder name.")
-    self.saveButtonIsEnabled.assertValues([false], "Remains disabled while form is incomplete.")
 
-    self.vm.inputs.creditCardChanged(cardDetails: ("4242 4242 4242 4242", 11, 99, "123"))
+    self.vm.inputs.creditCardChanged(cardDetails: ("4242 4242 4242 4242", 11, 99, "123", .visa))
+    self.saveButtonIsEnabled.assertValues([false], "Disabled form is incomplete")
     self.cardholderNameBecomeFirstResponder.assertValueCount(1, "Does not emit again.")
     self.paymentDetailsBecomeFirstResponder.assertValueCount(1, "Does not emit again.")
     self.vm.inputs.paymentInfo(isValid: true)
-    self.vm.inputs.cardBrand(isValid: true)
     self.vm.inputs.zipcodeChanged(zipcode: "123")
     self.saveButtonIsEnabled.assertValues([false, true], "Enabled when form is valid.")
   }
 
   func testSaveButtonEnabled() {
     self.saveButtonIsEnabled.assertDidNotEmitValue()
     self.vm.inputs.viewDidLoad()
+    self.vm.inputs.configure(with: .pledge, project: Project.template)
 
     self.vm.inputs.cardholderNameChanged("")
     self.saveButtonIsEnabled.assertDidNotEmitValue()
 
     self.vm.inputs.paymentInfo(isValid: false)
-    self.vm.inputs.cardBrand(isValid: false)
     self.saveButtonIsEnabled.assertDidNotEmitValue()
 
     self.vm.inputs.zipcodeChanged(zipcode: "")
-    self.saveButtonIsEnabled.assertValues([false], "Disabled form is incomplete")
 
     self.vm.inputs.cardholderNameChanged("Native Squad")
+    self.vm.inputs.creditCardChanged(cardDetails: ("4242 4242 4242 4242", 11, 99, "123", .visa))
+    self.saveButtonIsEnabled.assertValues([false], "Disabled form is incomplete")
     self.vm.inputs.paymentInfo(isValid: true)
-    self.vm.inputs.cardBrand(isValid: true)
     self.vm.inputs.zipcodeChanged(zipcode: "123")
 
     self.saveButtonIsEnabled.assertValues([false, true], "Enabled when form is valid.")
-
-    self.vm.inputs.paymentInfo(isValid: true)
-    self.vm.inputs.cardBrand(isValid: false)
-
-    self.saveButtonIsEnabled.assertValues([false, true, false], "Disabled if card brand is invalid")
```

Comment:
These assertions seem important. We need to verify that even if the form is valid (and the save button is enabled), if the user goes back and enters an invalid card brand, the save button needs to become disabled.

Reply:
Well cardBrand is no longer an input here. But I'll test the assertion with unavailable cards